core/prisma/seed/register-stub.js

```diff
@@ -0,0 +1,12 @@
+const Module = require("module");
+const path = require("path");
+
+const originalResolveFilename = Module._resolveFilename;
+
+Module._resolveFilename = function (request, parent, isMain, options) {
```

Comment:
Yes! This is necessary in order to stub out `React.cache`, `unstable_cache` and `server-only`. Basically before the script is run, this patches the way node resolves files, so that when it tries to load `next/cache`, or `react`, or any of the other things mentioned in that `if` statement, it gets the content of `stubs.js` instead, which defines most of those things as functions which return functions.